src/app/views/authentication/Authentication.tsx

```diff
@@ -38,69 +44,112 @@ const Authentication = (props: any) => {
       const authResponse = await authenticationWrapper.logIn();
       if (authResponse) {
         setLoginInProgress(false);
-        dispatch(getAuthTokenSuccess(!!authResponse.accessToken))
-        dispatch(getConsentedScopesSuccess(authResponse.scopes))
+        dispatch(getAuthTokenSuccess(!!authResponse.accessToken));
+        dispatch(getConsentedScopesSuccess(authResponse.scopes));
       }
     } catch (error) {
       const { errorCode } = error;
-      dispatch(setQueryResponseStatus({
-        ok: false,
-        statusText: messages['Authentication failed'],
-        status: errorCode === 'popup_window_error'
-          ? translateMessage('popup blocked, allow pop-up windows in your browser')
-          : errorCode ? errorCode.replace('_', ' ') : '',
-        messageType: MessageBarType.error
-      }));
+      dispatch(
+        setQueryResponseStatus({
+          ok: false,
+          statusText: messages['Authentication failed'],
+          status: getErrorAndHint(errorCode),
+          messageType: MessageBarType.error,
+        })
+      );
       setLoginInProgress(false);
       telemetry.trackException(
         new Error(errorTypes.OPERATIONAL_ERROR),
         SeverityLevel.Error,
         {
           ComponentName: componentNames.AUTHENTICATION_ACTION,
-          Message: `Authentication failed: ${errorCode ? errorCode.replace('_', ' ') : ''}`,
-        });
+          Message: `Authentication failed: ${
+            errorCode ? errorCode.replace('_', ' ') : ''
+          }`,
+        }
+      );
     }
   };
 
-  const showProgressSpinner = (): React.ReactNode => {
-    return <div className={classes.spinnerContainer}>
-      <Spinner className={classes.spinner} size={SpinnerSize.medium} />
-      {!minimised && <Label>
-        <FormattedMessage id={`Signing you ${loginInProgress ? 'in' : 'out'}...`} />
-      </Label>}
-    </div>;
+  const getErrorAndHint = (errorCode: string): string => {
```

Comment:
A function should do one thing only. Consider breaking down the functionalities here so that the `getErrorAndHint()` function gets the error with the hint (if a hint exists) and clearing the cache happens in an appropriately named function that is called elsewhere

Reply:
I agree. Let me break it down.